vulkaninfo/CMakeLists.txt

```diff
@@ -1,5 +1,81 @@
 # CMakeLists.txt file for building Vulkaninfo
 
+set(VULKANINFO_INCLUDE_DIRS
+    ${CMAKE_CURRENT_BINARY_DIR}
+    ${CMAKE_CURRENT_BINARY_DIR/..}
+   )
+
+if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
+    include(FindPkgConfig)
+    option(BUILD_WSI_XCB_SUPPORT "Build XCB WSI support" ON)
+    option(BUILD_WSI_XLIB_SUPPORT "Build Xlib WSI support" ON)
+    option(BUILD_WSI_WAYLAND_SUPPORT "Build Wayland WSI support" ON)
+    option(BUILD_WSI_MIR_SUPPORT "Build Mir WSI support" ON)
+    set(VULKANINFO_WSI_SELECTION "XCB" CACHE STRING "Select WSI target for vulkaninfo (XCB, XLIB, WAYLAND, MIR, DISPLAY)")
+
+    if (BUILD_WSI_XCB_SUPPORT)
+        find_package(XCB REQUIRED)
+    endif()
+
+    if (BUILD_WSI_XLIB_SUPPORT)
+        find_package(X11 REQUIRED)
+    endif()
+    
+    if (BUILD_WSI_WAYLAND_SUPPORT)
+        find_package(Wayland REQUIRED)
+    endif()
+
+    if (BUILD_WSI_MIR_SUPPORT)
+        find_package(Mir REQUIRED)
+    endif()
+
+    if (VULKANINFO_WSI_SELECTION STREQUAL "XCB")
+        if (NOT BUILD_WSI_XCB_SUPPORT)
+            message( FATAL_ERROR "Selected XCB for vulkaninfo build but not building Xcb support" )
+        endif()
+        set(VULKANINFO_INCLUDE_DIRS
+            ${XCB_INCLUDE_DIRS}
+            ${VULKANINFO_INCLUDE_DIRS}
+        )
+        link_libraries(${XCB_LIBRARIES})
```

Comment:
I know that this code was copied and pasted from someplace else, but it would be good to migrate towards Modern CMake practices here by using target_link_libraries() and target_compile_definitions() here.